### PR TITLE
Fix corrupted dashboard parameters to have "unnamed" names and slugs

### LINKED ISSUE #24500: Corrupted dashboard filter makes it impossible to view the question (unless removed)
**Describe the bug**
This is basically a duplicate of #15279, but that issue was created when we were still using the old syntax. Unfortunately, the E2E repro for this was more of a placeholder and it wasn't updated when the syntax changed so we didn't catch this regression.

**Logs**
```
POST /api/dashboard/*/dashcard/*/card/*/query results in

Output of dashboard->resolved-params does not match schema: 
	 [0;33m  {"af72ce9c" {:name (not ("Non-blank string" ""))}} [0m 
```

**To Reproduce**
1. Admin > Data Model > Sample Database > People > make sure to set **Source as "list of values"** and **Name as "search box"**
2. Simple question > Sample Dataset > People, and save as "Q1" and add to dashboard "D1"
3. On dashboard "D1", create two Text/Categories filters (both dropdown) - first connected (let's call it `List filter`) to Q1.Source, second (`Search filter`) connected to Q1.Name, save dashboard.
4. Question will not render and will display `There was a problem displaying this chart.`
5. `List filter` will not show a dropdown but "Enter some text" input field
6. `Search filter` will show "Search by name" but it doesn't seem it can find any value
7. Thanks to #16582 we can now remove the corrupted filter, so let's edit dashboard
8. Click on the gear icon next to the corrupted filter
image
9. You will notice it is possible to select question fields and to connect the filter to it (not sure if this should even be possible or should we just say "No fields available")
image
10. Remove filter from the sidebar (notice the question still doesn't display correctly) and save the dashboard
image
11. Either refresh or click on any of the working filters and set them, which will reset the corrupted question query
image

**Expected behavior**
Either the backend should make `name` required (and prevent adding a corrupted filter in the first place) or frontend should handle corrupted filter gracefully. @daltojohnso already made it possible to remove such filter in #16582. However, we should display the underlying question, especially since corrupted filter isn't even connected to the question.

**Screenshots**
image

**Information about your Metabase Installation:**
local dev, `master`, 3fc791200659436eadee74e2f2b12adc58f1dd9c, H2, Sample Database

**Severity**
I am not aware of the possibility to end up with a corrupted filter using the UI. It can only affect people trying to add filters via API and then forgetting to set the `name`.

**Additional context**
If filter cannot work without `name`, doesn't it make sense then to make that parameter required on the backend?

Why is it possible to send this payload (this adds corrupted filter to the dashboard)?
`PUT /api/dashboard/:id`
```json
{
  "name": "",
  "slug": "",
  "id": "af72ce9c",
  "type": "string/=",
  "sectionId": "string"
}
```

### PR DESCRIPTION
**Pull request by calherries**
Fixes #24500.

Sets the dashboard parameters with empty string `name` or `slug` to have `"unnamed"` for both.

### CODE REVIEW

File: e2e/test/scenarios/dashboard-filters/reproductions/24500-gracefully-deal-with-corrupted-filter.cy.spec.js [hunk 1]
```diff
@@ -110,17 +110,23 @@ describe.skip("issues 15279 and 24500", () => {
...
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("Select…");
-
     editDashboard();
-    filterWidget().last().find(".Icon-gear").click();
-    // Uncomment the next line if we end up disabling fields for the corrupted filter
-    // cy.findByText("No valid fields")
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("Remove").click();
+    cy.findByTestId("edit-dashboard-parameters-widget-container")
```
Comment (calherries) [inline e2e/test/scenarios/dashboard-filters/reproductions/24500-gracefully-deal-with-corrupted-filter.cy.spec.js:114]: `filterWidget` doesn't get the filter widget in edit mode. Maybe it worked before, but it doesn't now.

File: e2e/test/scenarios/dashboard-filters/reproductions/24500-gracefully-deal-with-corrupted-filter.cy.spec.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ const questionDetails = {
 
 const dashboardDetails = { parameters };
 
-describe.skip("issues 15279 and 24500", () => {
+describe("issues 15279 and 24500", () => {
```
Comment (calherries) [inline e2e/test/scenarios/dashboard-filters/reproductions/24500-gracefully-deal-with-corrupted-filter.cy.spec.js:47]: This is the repro that was created for https://github.com/metabase/metabase/issues/24500

**Review by qnkhuat — CHANGES_REQUESTED** [review]
I don't think this is gonna fix #24500.

One problem from the issue is that if you have 2 filters without a name, then setting value for one parameter will populate the value to both filters after refresh. It's probably because FE gets the value from URL `?unamed={{some_value}}`.

I think we need to make sure that the name and slug we assoc in the after-select is unique to fix that.

**Review by ranquild — APPROVED** [review]
FE changes ok

**Comment by calherries** [comment]
@qnkhuat 

> I don't think this is gonna fix https://github.com/metabase/metabase/issues/24500.

Tell me why. It fixes the repro.

> One problem from the issue is that if you have 2 filters without a name, then setting value for one parameter will populate the value to both filters after refresh. It's probably because FE gets the value from URL ?unamed={{some_value}}.

That's a different issue. I'm solving it with [this](https://github.com/metabase/metabase/issues/27653#issuecomment-1746565164) approach.

**Review by qnkhuat — APPROVED** [review]
I guess the expected behavior of #24500 is unclear, but okay I think it's an improvement to not error out from BE.

and I'm okay with solving the issue of unique parameter names separately.

**Pull request merged**